hack/ipa_debug_tools.sh

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env bash
+
+# The path to the directory that holds this script
+CURRENT_SCRIPT_DIR="$(dirname -- "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+# This ceates a minimal set up in /tmp/debug-initramfs directory for
+# comparing built ipa initramfs the master ipa initramfs.
+set_up_debug_dirs() {
+    # Export DISABLE_UPLOAD, ENABLE_BOOTSTRAP_TEST, TEST_IN_CI to allow local
+    # test run.
+    export DISABLE_UPLOAD="true"
+    export ENABLE_BOOTSTRAP_TEST="false"
+    export TEST_IN_CI="false"
+
+    # Build ipa with build_ipa.sh
+    cd "$CURRENT_SCRIPT_DIR/../jenkins/scripts/dynamic_worker_workflow" || exit
+    if [[ ! -x ./build_ipa.sh ]]; then
+        echo "Error: build_ipa.sh not found or not executable in $(pwd)"
+        exit 1
+    fi
+    ./build_ipa.sh
+
+    # Even if build_ipa.sh exits with error this script can continue as long as
+    # /tmp/dib/ironic-python-agent.initramfs has been created
+    if [[ ! -f /tmp/dib/ironic-python-agent.initramfs ]]; then
+        exit 1
+    fi
+
+    # Unzip ironic-python-agent.initramfs
+    sudo mkdir -p /tmp/debug-initramfs/build-ipa-initramfs
+    sudo cp /tmp/dib/ironic-python-agent.initramfs /tmp/debug-initramfs/
+    cd /tmp/debug-initramfs/build-ipa-initramfs || exit
+    gunzip -c ../ironic-python-agent.initramfs | sudo cpio -id
+
+    # Get master ironic-python-agent.initramfs and unzip
+    sudo mkdir -p /tmp/debug-initramfs/master-ipa-initramfs
+    cd /tmp/debug-initramfs || exit
+    sudo wget https://tarballs.opendev.org/openstack/ironic-python-agent/dib/ipa-centos9-master.tar.gz
+    sudo tar -xzf ipa-centos9-master.tar.gz
+    cd master-ipa-initramfs || exit
+    gunzip -c ../ipa-centos9-master.initramfs | sudo cpio -id
+}
+
+# -----------------------------------------------------------------------------
+# compare_dir_sizes <relative_path> [base_dir1] [base_dir2]
+#
+# Compares the sizes of files and directories at the given <relative_path>
+# inside two extracted initramfs directories (or any two directories).
+#
+# Usage examples:
+#   ./ipa_debug_tools.sh compare_dir_sizes
+#   ./ipa_debug_tools.sh compare_dir_sizes usr
+#   ./ipa_debug_tools.sh compare_dir_sizes usr/lib ~/ipa-build-initramfs ~/ipa-initramfs
+#   ./ipa_debug_tools.sh compare_dir_sizes . ~/ipa-build-initramfs ~/ipa-initramfs
+#
+# - <relative_path>: Path inside the compared directories (e.g., usr, usr/lib)
+# - [base_dir1]:     First base directory (default: /tmp/debug-initramfs/build-ipa-initramfs)
+# - [base_dir2]:     Second base directory (default: /tmp/debug-initramfs/master-ipa-initramfs)
+#
+# The function prints a table of entries, their sizes in both directories,
+# and the size difference. It also lists files only present in one directory.
+# -----------------------------------------------------------------------------
+compare_dir_sizes() {
+    path="${1:-}"
+    basedir1="${2:-/tmp/debug-initramfs/build-ipa-initramfs}"
```

Comment:
```suggestion
    build_dir="${2:-/tmp/debug-initramfs/build-ipa-initramfs}"
```
More descriptive vars please